File: Electronic store/c#/src/Services/FileManager.cs
namespace ElectronicStoreSystem.Services;

public static class FileManager
{
    public static void EnsureFileExists(string filePath)
    {
        string? directory = Path.GetDirectoryName(filePath);
        if (!string.IsNullOrEmpty(directory) && !Directory.Exists(directory))
        {
            Directory.CreateDirectory(directory);
        }
        if (!File.Exists(filePath))
        {
            using (File.Create(filePath)) { }
        }
    }

    public static List<string> ReadAllLines(string filePath)
    {
        EnsureFileExists(filePath);
        return File.ReadAllLines(filePath)
                   .Where(line => !string.IsNullOrWhiteSpace(line))
                   .ToList();
    }

    public static void AppendLine(string filePath, string line)
    {
        EnsureFileExists(filePath);
        File.AppendAllLines(filePath, new[] { line });
    }

    public static void WriteAllLines(string filePath, List<string> lines)
    {
        EnsureFileExists(filePath);
        File.WriteAllLines(filePath, lines);
    }
}

File: Electronic store/python/src/file_manager.py
from pathlib import Path
from typing import List

def ensure_file_exists(file_path: Path) -> None:
    file_path.parent.mkdir(parents=True, exist_ok=True)
    if not file_path.exists():
        file_path.touch()

def read_all_lines(file_path: Path) -> List[str]:
    ensure_file_exists(file_path)
    with open(file_path, "r", encoding="utf-8") as f:
        return [line.strip() for line in f if line.strip()]

def append_line(file_path: Path, line: str) -> None:
    ensure_file_exists(file_path)
    with open(file_path, "a", encoding="utf-8") as f:
        f.write(f"{line}\n")

def write_all_lines(file_path: Path, lines: List[str]) -> None:
    ensure_file_exists(file_path)
    with open(file_path, "w", encoding="utf-8") as f:
        for line in lines:
            f.write(f"{line}\n")